### DISCUSSION TITLE
reactFlow.fitView with offset due to opened sidebar

### DISCUSSION THREAD

**Discussion by T005206** [xyflow/xyflow#4366] [Layouting] [answered]
Hi, How can I center a node or several nodes using reactFlow.fitView({ nodes: nodesToCenter }); but taking into account a displacement due to a sidebar that opens on the right of which I have the size in pixels?

Thanks

**Comment by bcakmakoglu** [comment]
`fitView` has an option for `offset` which can be an object wit `x: number, y: number` - maybe that works for you?

**Reply by T005206** [reply]
Thank @bcakmakoglu but fitview option don't have this property. I'm in "reactflow": "11.9.3",
![image](https://github.com/xyflow/xyflow/assets/127098235/9ee3b187-2701-448b-9654-5b698139af5c)

**Reply by bcakmakoglu** [reply]
mhhh.... I thought it did :(
Maybe I'm imagining things then, that's stupid :/

**Reply by T005206** [reply]
anyway thanks !!!

**Reply by bcakmakoglu** [reply]
Yeah my bad, *VueFlow* has this option but ReactFlow doesn't

**Reply by bcakmakoglu** [reply]
You could try doing your own `fitView` by using this

```ts
const bounds = getNodesBounds(nodes, nodeOrigin);

const { x, y, zoom } = getViewportForBounds(
  bounds,
  width,
  height,
  options.minZoom ?? minZoom,
  options.maxZoom ?? maxZoom,
  options.padding ?? 0.1
);
```

and using `setViewport` to set it, that *might* work 🤔

**Reply by T005206** [reply, marked as answer]
Thanks finally i change the .react-flow__renderer element to fit the 100% with - sidebarwidth and works 

```js
const sidebarElement = document.querySelector<HTMLDivElement>('[data-pc-name="sidebar"]');
if (sidebarElement && sidebarState.visible) { setSidebarWidth(sidebarElement.offsetWidth); } else setSidebarWidth(0); const reactFlowRenderer = document.querySelector<HTMLDivElement>('.react-flow__renderer'); if (reactFlowRenderer) reactFlowRenderer.style.width =calc(100% - ${sidebarWidth}px); 
const reactFlowRenderer = document.querySelector<HTMLDivElement>('.react-flow__renderer');
if (reactFlowRenderer) reactFlowRenderer.style.width = `calc(100% - ${sidebarWidth}px)`;
```


Thansk for your replies
